src/Modal.js

```diff
@@ -106,7 +85,8 @@ export class Modal extends Component {
       backdropStyle,
       ...otherProps
     } = this.props;
-    const { deviceHeight, deviceWidth, isVisible } = this.state;
+    const { isVisible } = this.state;
+    const { height, width } = useWindowDimensions();
```

Comment:
I don't think this will work because [hooks can't be used in class components](https://reactjs.org/docs/hooks-faq.html#:~:text=You%20can't%20use%20Hooks,implementation%20detail%20of%20that%20component.).